fabric-item-api-v1/src/testmod/java/net/fabricmc/fabric/test/item/CustomDamageTest.java

```diff
@@ -76,5 +79,13 @@ public ItemStack getRecipeRemainder(ItemStack stack) {
 
 			return ItemStack.EMPTY;
 		}
+
+		@Override
+		public @Nullable FoodComponent getFoodComponent(ItemStack stack) {
+			return new FoodComponent.Builder()
+					.hunger(20 - 20 * stack.getDamage() / stack.getMaxDamage())
+					.saturationModifier(0.5f)
+					.build();
+		}
```

Comment:
This is in the custom damage test. I think it would be best to move out into its own file.

Reply:
I just can't think up other existing NBT for the moment so directly use the most common NBT, damage. Maybe I could use enchantments.

Reply:
Ok, thats fine maybe just put a comment on the method saying its for the stackaware food test.